components/raftstore/src/store/peer.rs

```diff
@@ -2550,7 +2567,40 @@ where
                 // Update apply index to `last_applying_idx`
                 self.read_progress
                     .update_applied(self.last_applying_idx, &ctx.coprocessor_host);
-                self.notify_leader_the_peer_is_available(ctx);
+                if self.wait_data {
+                    self.notify_leader_the_peer_is_available(ctx);
```

Comment:
what will happen if notify_leader_the_peer_is_available failed?

Reply:
After the leader has applied the non-witness switch, it will register a tick. If it has not received the non-witness push information for a long time, it will take the initiative to obtain the information.